controllers/humiocluster_controller.go

```diff
@@ -1922,6 +1928,64 @@ func (r *HumioClusterReconciler) ensureMismatchedPodsAreDeleted(ctx context.Cont
 
 	podsForDeletion := desiredLifecycleState.podsToBeReplaced
 
+	pdbSpec := hnp.GetPodDisruptionBudget()
+	r.Log.Info("Entering PDB enforcement check", "nodePool", hnp.GetNodePoolName(), "pdbSpec", pdbSpec, "pdbSpec.Enabled", pdbSpec != nil && pdbSpec.Enabled)
```

Comment:
As far as I can tell, we don't really need to add anything to `ensureMismatchedPodsAreDeleted`, since `ensureMismatchedPodsAreDeleted` is handling the case of pod replacements due to config changes and version changes/upgrades.

The case for adding PDB is more about ensuring components "outside the humio-operator" e.g. k8s worker node drain won't bring more pods/instances down than the configured amount. Config changes and version changes/upgrades is already controlled through the configurable update strategy and this is already included in the current master version of `ensureMismatchedPodsAreDeleted`.

Let me know if I missed something. Thank you.

Reply:
Makes total sense you explained. So what I understand what you said is that `ensureMismatchedPodsAreDeleted` don't require explicit PDB handling because the function already coordinates pod replacements using:
- The cluster's update strategy (rolling updates)
- Max unavailable pods calculation
- Readiness checks
- Version/config mismatch detection

And PDB's are primarily for external kubernetes-level disruption, like node drains or maintenance, rather than operator-controlled replacements.

Said that, I have removed the PDB section from `ensureMismatchedPodsAreDeleted`